src/System.Text.Json/tests/JsonDocumentTests.cs

```diff
@@ -492,10 +492,28 @@ public static void ParseJson_SequenceBytes_Multi(bool compactData, TestCaseType
 
             byte[] dataUtf8 = Encoding.UTF8.GetBytes(jsonString);
 
+            var buffer = new ArrayBufferWriter<byte>(1024);
+            var expectedBuffer = new ArrayBufferWriter<byte>(1024);
+            var options = new JsonWriterOptions
+            {
+                Indented = !compactData
+            };
+
+            var writer = new Utf8JsonWriter(buffer, options);
+            var expectedWriter = new Utf8JsonWriter(expectedBuffer, options);
+
             using (JsonDocument doc = stringDocBuilder?.Invoke(jsonString) ?? bytesDocBuilder?.Invoke(dataUtf8))
             {
                 Assert.NotNull(doc);
 
+                doc.Write(writer);
+                writer.Flush();
+
+                doc.RootElement.WriteValue(expectedWriter);
+                expectedWriter.Flush();
```

Comment:
Please don't mix write tests in with the read tests; they should be new tests altogether.